sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -186,12 +188,12 @@ internal TableClient(Uri endpoint, string tableName, TableSharedKeyPipelinePolic
             HttpPipeline pipeline = sasCredential switch
```

Comment:
Should we consider splitting out the common parts to simplify the switch a bit?   Maybe something like:

```csharp
HttpPipelinePolicy authPolicy = sasCredential switch
{
    null => policy,
    _ => new AzureSasCredentialSynchronousPolicy(sasCredential)
};

HttpPipeline pipeline = HttpPipelineBuilder.Build(options, perCallPolicies: perCallPolicies, perRetryPolicies: new[] { authPolicy }, new ResponseClassifier());
```